build_resolvers/lib/src/build_asset_uri_resolver.dart

```diff
@@ -191,8 +191,27 @@ class BuildAssetUriResolver extends UriResolver {
   }
 
   @override
-  Uri? restoreAbsolute(Source source) =>
-      lookupCachedAsset(source.uri)?.uri ?? source.uri;
+  // ignore: override_on_non_overriding_member
+  Uri pathToUri(String path) {
+    var pathSegments = resourceProvider.pathContext.split(path);
```

Comment:
It looks like we always use the `posix` context to create these paths. We should either use the resource providers context there as well or just use posix directly here probably?

Reply:
There is some contradiction here. The `ResourceProvider` and its `pathContext` should correspond to the style of path(s). If we always build Posix path(s), then `pathContext` must be Posix. Otherwise I don't understand why it would work.

Reply:
It seems reasonable to me to update to use the `pathContext` when constructing the paths, do you mind updating that here as well?

Reply:
Hm... We also use `posix` in `assetPath`, and we don't have access to `ResourceProvider`. In this case, it is better to use `posix` directly here as well.

Reply:
sgtm